includes/api/class-wc-admin-rest-leaderboards-controller.php

```diff
@@ -0,0 +1,481 @@
+		}
+
+		return array(
+			'id'      => 'coupons',
+			'label'   => __( 'Top Coupons - Number of Orders', 'woocommerce-admin' ),
+			'headers' => array(
+				array(
+					'label' => __( 'Coupon Code', 'woocommerce-admin' ),
+				),
+				array(
+					'label' => __( 'Orders', 'woocommerce-admin' ),
+				),
+				array(
+					'label' => __( 'Amount Discounted', 'woocommerce-admin' ),
+				),
+			),
+			'rows'    => $rows,
+		);
+	}
+
+	/**
+	 * Get the data for the categories leaderboard.
+	 *
+	 * @param int    $rows Number of rows.
+	 * @param string $after Items after date.
+	 * @param string $before Items before date.
+	 * @param string $persisted_query URL query string.
+	 */
+	public function get_categories_leaderboard( $rows, $after, $before, $persisted_query ) {
+		$categories_data_store = new WC_Admin_Reports_Categories_Data_Store();
+		$categories_data       = $categories_data_store->get_data(
+			array(
+				'orderby'       => 'items_sold',
+				'order'         => 'desc',
+				'after'         => $after,
+				'before'        => $before,
+				'extended_info' => true,
+			)
+		);
+
+		$rows = array();
+		foreach ( $categories_data->data as $category ) {
+			$url_query     = wp_parse_args(
+				array(
+					'filter'     => 'single_category',
+					'categories' => $category['category_id'],
+				),
+				$persisted_query
+			);
+			$category_url  = wc_admin_url( 'analytics/categories', $url_query );
+			$category_name = $category['extended_info'] && $category['extended_info']['name'] ? $category['extended_info']['name'] : '';
+			$rows[]        = array(
+				array(
+					'display' => "<a href='{$category_url}'>{$category_name}</a>",
+					'value'   => $category_name,
+				),
+				array(
+					'display' => wc_admin_number_format( $category['items_sold'] ),
+					'value'   => $category['items_sold'],
+				),
+				array(
+					'display' => wc_price( $category['net_revenue'] ),
+					'value'   => $category['net_revenue'],
+				),
+			);
+		}
+
+		return array(
+			'id'      => 'categories',
+			'label'   => __( 'Top Categories - Items Sold', 'woocommerce-admin' ),
+			'headers' => array(
+				array(
+					'label' => __( 'Category', 'woocommerce-admin' ),
+				),
+				array(
+					'label' => __( 'Items Sold', 'woocommerce-admin' ),
+				),
+				array(
+					'label' => __( 'Net Revenue', 'woocommerce-admin' ),
+				),
+			),
+			'rows'    => $rows,
+		);
+	}
+
+	/**
+	 * Get the data for the customers leaderboard.
+	 *
+	 * @param int    $rows Number of rows.
+	 * @param string $after Items after date.
+	 * @param string $before Items before date.
+	 * @param string $persisted_query URL query string.
+	 */
+	public function get_customers_leaderboard( $rows, $after, $before, $persisted_query ) {
+		$customers_data_store = new WC_Admin_Reports_Customers_Data_Store();
+		$customers_data       = $customers_data_store->get_data(
+			array(
+				'orderby' => 'total_spend',
+				'order'   => 'desc',
+				'after'   => $after,
+				'before'  => $before,
+			)
```

Comment:
Can this use `per_page`?

Also I'm not even sure the date range is being honored by `WC_Admin_Reports_Customers_Data_Store` - changing the date range doesn't seem to change the response.

Reply:
`per_page` works, but you're right, the date range is not respected by the customers data store.

Do you think this is worth opening a separate issue to add in date querying?  The store joins the orders table, so it shouldn't add too much overhead.

Reply:
> Do you think this is worth opening a separate issue to add in date querying? The store joins the orders table, so it shouldn't add too much overhead.

Definitely!